src/utils/lru_cache.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <list>
+#include <unordered_map>
+#include <utility>
+
+namespace memgraph::utils {
+
+template <class TKey, class TVal>
+class LRUCache {
+ public:
+  LRUCache(int cache_size_) : cache_size(cache_size_){};
+
+  void put(const TKey &key, const TVal &val) {
+    auto it = item_map.find(key);
+    if (it != item_map.end()) {
+      item_list.erase(it->second);
+      item_map.erase(it);
+    }
```

Comment:
Why are you cleaning if there was an element found, and then push again?

Reply:
That way we will insert element at the back of the list again, because we are not sorting elements explicitly, but appending LRU element at the back of the list. So that way we will "mark" that element as LRU.

Reply:
I didn't realize at first that comment is about put method, not get. We will use remove part of code in put method if we want to change value for the key that is already in the cache.